src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -237,6 +237,11 @@ protected override async Task WriteCacheBytesAsync(string cacheKey, byte[] bytes
             if (cacheSerializerHints != null && cacheSerializerHints?.SuggestedCacheExpiry != null)
             {
                 cacheExpiry = cacheSerializerHints.SuggestedCacheExpiry.Value.UtcDateTime - DateTime.UtcNow;
+                if (cacheExpiry < TimeSpan.Zero)
```

Comment:
Hmm, should you not delete the cache entry instead?

Reply:
we could but then the L1/L2 will be out of sync.